src/snowflake/connector/network.py

```diff
@@ -980,14 +981,9 @@ def handle_invalid_certificate_error(self, conn, full_url, cause) -> None:
     def _handle_unknown_error(self, method, full_url, headers, data, conn) -> None:
         """Handles unknown errors."""
         if data:
-            try:
-                decoded_data = json.loads(data)
-                if decoded_data.get("data") and decoded_data["data"].get("PASSWORD"):
-                    # masking the password
-                    decoded_data["data"]["PASSWORD"] = "********"
-                    data = json.dumps(decoded_data)
-            except Exception:
-                logger.info("data is not JSON")
+            _, masked_data, err_str = SecretDetector.mask_secrets(data)
```

Comment:
Should we also check the first returned value to see if the data is masked? Or is this more like a best effort?

Reply:
if data is not masked, then original data is returned. It basically means that no secrets were detected. I'm not sure how we would use the `masked` value

Reply:
I see. Thanks for the explanation :)